### PR TITLE
Adding Run() methods to RomManager

### PR DESCRIPTION
**Pull request by Rbravo555**
**📝 Description**
So far, the RomManager allowed to **train** and **test** a reduced order model. However, we were missing a method in the RomManager for cheaply launching the generated models.
 
This PR adds the methods in the RomManager for running a set of simulations using either of the models. Clearly, if any of the required files for launching them is missing, the simulations will fail, as no training or snapshots matrices storage is considered. 

In order not to generate any conflict with currently open PRs, this PR is not tackling known shortcomings of the current approach, for example if a user wants to launch a ROM using a different amount of modes each time, she would have to modify the RomParameters.json directly. That is, not all of the ROM simulation parameters are controlled from the RomManager and a few are still relying on the RomParameters.json. A possible solution would necessarily involve modifying the RomAnalysis class to pass the RomParameters object instead of reading it from disk. This or other solutions will be deffered to future PRs.

**🆕 Changelog**
Added six methods to the RomManager.

Methods to be called from the instantiated RomManager 
- RunFOM(self, mu_run=None)
- RunROM(self, mu_run=None)
- RunHROM(self, mu_run=None)

Auxiliary methods inside the RomManager class:
- LaunchRunFOM(self, mu_run)
- LaunchRunROM(self, mu_run)
- LaunchRunHROM(self, mu_run)

The branch is compatible with the RomManager example in the Examples repo [here](https://github.com/KratosMultiphysics/Examples/tree/master/rom_application/RomManager)

### CODE REVIEW

**Comment by SADPR** [comment]
It looks good to me. I would wait for @rubenzorrilla 's opinion to merge it. But it is something very useful.

**Review by NicolasSR — COMMENTED** [review]
Very nice addition. Seems fine to me

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]:
```suggestion
        if not mu_run:
```
I think it's more pythonic this way.

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]:
```suggestion
```
Avoid extra whitespaces.

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]: Why assigning a value to it? Wouldn't it be better to keep the `None` default value?

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]: (same comments)

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py:171]:
```suggestion
            self._ChangeRomFlags(simulation_to_run = "PG")
        else:
            err_msg = f"Provided projection strategy '{self.general_rom_manager_parameters["projection_strategy"].GetString()}' is not supported. Available options are 'galerkin', 'lspg' and 'petrov_galerkin'."
            raise Exception(err_msg)
```

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]:
```suggestion
```
Avoid extra whitespaces.

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]: I'd throw the same error message I suggest for `RunROM`.

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]: (same comment)

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]:
```suggestion
```

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]:
```suggestion
```

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]:
```suggestion
    def __LaunchRunFOM(self, mu_run):
```
As far as I see these can be mangled.

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]:
```suggestion
    def __LaunchRunROM(self, mu_run):
```

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]:
```suggestion
    def __LaunchRunHROM(self, mu_run):
```

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (Rbravo555) [reply applications/RomApplication/python_scripts/rom_manager.py]: In the past, when using `not mu_run`, sometime I came across the error 
    ValueError: The truth value of an array with more than one element is ambiguous.
On the other hand, `mu_run is None` always works independently of the type of mu_run

I must recognize that given the type of `mu` (either a Python list or None), the suggested code will always work. 

I would stick to the less Pythonic, yet more English and robust expression. But if you insist I will change it across the script

File: applications/RomApplication/python_scripts/rom_manager.py
Comment (Rbravo555) [reply applications/RomApplication/python_scripts/rom_manager.py]: If the user doesn't  pass a list of parameters in `mu_run`, that means that a single case will be run with the parameters already contained in the Project Parameters.json. That is exactly the string we add to `mu_run`. Why we add this string? because later on (in `LaunchRunFOM`), we loop over the elements of `mu_run`, and if `mu_run` is None, we would have an error at

        for Id, mu in enumerate(mu_run):
            parameters = self.UpdateProjectParameters(parameters, mu)
            parameters = self._StoreResultsByName(parameters,'FOM_Run',mu,Id)
            model = KratosMultiphysics.Model()
            analysis_stage_class = self._GetAnalysisStageClass(parameters)
            simulation = self.CustomizeSimulation(analysis_stage_class,model,parameters)
            simulation.Run()

It's not too bad a trick, I believe.

Comment (rubenzorrilla) [reply applications/RomApplication/python_scripts/rom_manager.py]: I think it is a bit hacky. What if we default as `mu_run = [None]`?

Comment (Rbravo555) [reply applications/RomApplication/python_scripts/rom_manager.py]: This one I like more. That would also solve the comment above

**Comment by Rbravo555** [comment]
I have considered all suggestions, thank you all. Now I only wait for green light to merge @rubenzorrilla

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**
